app/script/components/copyToClipboard.js

```diff
@@ -0,0 +1,47 @@
+/*
+ * Wire
+ * Copyright (C) 2018 Wire Swiss GmbH
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ *
+ */
+
+'use strict';
+
+window.z = window.z || {};
+window.z.components = z.components || {};
+
+z.components.CopyToClipboard = class CopyToClipboard {
+  constructor({text, trigger}) {
+    this.text = text;
+
+    trigger(() => z.util.ClipboardUtil.copyText(document.getElementById('copy-to-clipboard__source').innerText));
+  }
+
+  onClick() {
+    if (window.getSelection) {
+      const selectionRange = document.createRange();
+      selectionRange.selectNode(document.getElementById('copy-to-clipboard__source'));
```

Comment:
Humm when might need to have dynamic ids. If there are 2 `copy-text` component in the page, the selector will always select the first element (and you might end up copying the wrong text)

And idea could be let `ko` inject the root element of the component in the constructor and then select by a class only in that root DOM element.
Using the `createViewModel factory function` method would be to right way to do so I believe http://knockoutjs.com/documentation/component-registration.html